py/server/deephaven/table.py

```diff
@@ -55,13 +60,282 @@
+        j_format_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFormatOperationsRecorder)
+        return self.__class__(j_format_ops_recorder.formatRowWhere(cond, formula))
+
+    def format_column_where(self, col: str, cond: str, formula: str):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.format_column_where` operation applied to
+        nodes."""
+        j_format_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFormatOperationsRecorder)
+        return self.__class__(j_format_ops_recorder.formatColumnWhere(col, cond, formula))
+
+
+class _SortOperationsRecorder(Protocol):
+    """A mixin for creating sort operations to be applied to individual nodes of either RollupTable or
+    TreeTable."""
+
+    def sort(self, order_by: Union[str, Sequence[str]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.sort` operation applied to nodes."""
+        order_by = to_sequence(order_by)
+        j_sort_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JSortOperationsRecorder)
+        return self.__class__(j_sort_ops_recorder.sort(order_by))
+
+    def sort_descending(self, order_by: Union[str, Sequence[str]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.sort_descending` applied to nodes."""
+        order_by = to_sequence(order_by)
+        j_sort_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JSortOperationsRecorder)
+        return self.__class__(j_sort_ops_recorder.sortDescending(order_by))
+
+
+class _FilterOperationsRecorder(Protocol):
+    """A mixin for creating filter operations to be applied to individual nodes of either RollupTable or
+    TreeTable."""
+
+    def where(self, filters: Union[str, Filter, Sequence[str], Sequence[Filter]]):
+        """Returns a new recorder with the :meth:`~deephaven.table.Table.where` operation applied to nodes."""
+        filters = to_sequence(filters)
+        j_filter_ops_recorder = jpy.cast(self.j_node_ops_recorder, _JFilterOperationsRecorder)
+        return self.__class__(j_filter_ops_recorder.where(filters))
+
+
+class RollupNodeOperationsRecorder(JObjectWrapper, _FormatOperationsRecorder,
+                                   _SortOperationsRecorder):
+    """Recorder for node-level operations to be applied when gathering snapshots of RollupTable. Supported operations
+    include column formatting and sorting.
+
+    Note: It should not be instantiated directly. User code must call :meth:`~RollupTable.node_operation_recorder` to
+    create an instance of the recorder.
+    """
+
+    j_object_type = _JRollupTableNodeOperationsRecorder
+
+    @property
+    def j_object(self) -> jpy.JType:
+        return self.j_node_ops_recorder
+
+    def __init__(self, j_node_ops_recorder: jpy.JType):
+        self.j_node_ops_recorder = j_node_ops_recorder
+
+
+class RollupTable(JObjectWrapper):
+    """ A RollupTable is generated as a result of applying the :meth:`~deephaven.table.Table.rollup` operation on a
```

Comment:
The pydoc says how it is created but not what it is.  A user needs some sort of conceptual framework for understanding the object/class.